scalalib/src/GenIdeaImpl.scala

```diff
@@ -226,12 +226,16 @@ object GenIdeaImpl {
     def sbtLibraryNameFromPom(pom : os.Path) : String = {
       val xml = scala.xml.XML.loadFile(pom.toIO)
 
-      val groupId = (xml \ "groupId").text
+      val parent = xml \ "parent"
       val artifactId = (xml \ "artifactId").text
-      val version = (xml \ "version").text
+      val groupId = Some(xml \ "groupId").filter(_.nonEmpty).getOrElse(parent \ "groupId").text
+      val version = Some(xml \ "version").filter(_.nonEmpty).getOrElse(parent \ "version").text
 
-      // The scala version here is non incidental
-      s"SBT: $groupId:$artifactId:$version:jar"
+      val artifactWithScalaVersion = artifactId.substring(artifactId.length - 5) match {
```

Comment:
Forget about the suggested val name, I misread.
My concern is primary the explicit versions. What about 2.13?

Reply:
Yeah, I don't like it anymore than you do

The naming that Idea expects is very strange. I will add some examples/tests in order to clarify.

I ended up the writing the code that checks for the Scala version suffix that way because it mimics the way that the Idea Scala Plugin actually does it. I think it was a bad decision.

About appending "_2.12", Idea expects the version to match mill's version of Scala. I wanted to use the BuildInfo plugin for that, but I am not sure if mill itself is using it. I saw the version already hardcoded in the code, so I went with it. I will try to use the BuildInfo plugin instead.